### DISCUSSION TITLE
Custom attribute when importing objects from JSON

### DISCUSSION THREAD

**Discussion by heaven** [fabricjs/fabric.js#6944] [Q&A] [answered]
We do store the drawing in the database, the data model is pretty simple a Whiteboard Session and Whiteboard Object. To save the data we dump the whole canvas to JSON and put it in the database. Every Object in the JSON is saved as a separate record in the database. The problem is, when there are many of these on the whiteboard API calls become slower and slower. As a solution, we wanted to update each object separately and only saving new/updated/deleted objects.

I was wondering if there's something in the Fabrick library that can help us. At a minimum would be great to have the ability to add a custom attribute to each imported object that'd survive all the edits till the next export. So this way we can save our internal `object_id` and then map it to the existing record in the database.

**Comment by asturur** [comment, marked as answer]
hi @heaven, this question has been asked already a couple of time, i think you should find a decent answer looking for old issues / disucssions of people asking to add an ID to the serialized canvas.

https://github.com/fabricjs/fabric.js/issues/6917

check if some of those linked resources helps you.

**Reply by heaven** [reply]
Thank you, @asturur.
